client/my-sites/people/people-profile/index.jsx

```diff
@@ -1,31 +1,46 @@
 /**
  * External dependencies
  */
-import { connect } from 'react-redux';
-import React, { Component, Fragment } from 'react';
+import React from 'react';
 import { localize } from 'i18n-calypso';
 import classNames from 'classnames';
 import { get } from 'lodash';
 import { recordTrack } from 'calypso/reader/stats';
 import page from 'page';
 import { decodeEntities } from 'calypso/lib/formatting';
+import { useEffect, useState } from '@wordpress/element';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
 import Gravatar from 'calypso/components/gravatar';
 import InfoPopover from 'calypso/components/info-popover';
 import { withLocalizedMoment } from 'calypso/components/localized-moment';
-import { requestExternalContributors } from 'calypso/state/data-getters';
+import useExternalContributorsQuery from 'calypso/data/external-contributors/use-external-contributors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
 /**
  * Style dependencies
  */
 import './style.scss';
 
-class PeopleProfile extends Component {
-	getRole = () => {
-		const { invite, user } = this.props;
+const PeopleProfile = ( { siteId, type, user, invite, moment, translate } ) => {
+	const [ isExternalContributor, setIsExternalContributor ] = useState( false );
+	const { data: externalContributors } = useExternalContributorsQuery( siteId );
+
+	useEffect( () => {
+		if (
+			externalContributors &&
+			user?.ID &&
+			externalContributors.includes( user?.linked_user_ID ?? user.ID )
+		) {
+			setIsExternalContributor( true );
+		}
+	}, [ user, externalContributors ] );
```

Comment:
There doesn't need to be an effect or state here. We're merely computing a value from props:
```js
const isExternalContributor = externalContributors && externalContributors.includes( user?.linked_user_ID ?? user?.ID );
```
That can be done right in the render function.